### PR TITLE
Updating the setup script to handle sample data import for Docker 

### LINKED ISSUE #2270: Setup script doesn't handle Docker intuitively
**Describe the bug**

1. The setup script doesn't
    1. remember whether the user running the script previously answered `Y` to this question:
        ```
        Are you setting up this project using Docker? (y/N)
        ```
    1. doesn't detect whether docker is running the application and if so doesn't prompt the user to import the sample data. This means that the user has to refer to INSTALLATION.md and manually do the importation. This is not a good user experience.

**To Reproduce**

1. Run the setup script, select Docker
1. You never get prompted for data import if docker is running the application
1. The script doesn't store your previous Docker related answers

**Expected behavior**

1. Run the setup script, select Docker
1. Get prompted for data import if docker is running the application
1. The script stores your previous Docker related answers
1. Sample data is correctly imported
1. Update `INSTALLATION.md` to:
    1.  let people know that after starting docker for the first time that they should run the installation script again
    1. show people how to install sample data from the CLI when docker is being used.
1. There must be a docker specific way to manually import data successfully with appropriate errors if it fails.
1. The existing non-docker data importation script must fail with an appropriate error if the non-docker API instance and database is not. It does not do that now. Docker can be running and the non-docker data importation runs without failing or giving appropriate messaging.
1. Tests are written to ensure the script always has the newly desired behavior

**Actual behavior**

- See above

**Screenshots**

image

**Additional details**

- N/A

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 

- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by VanshikaSabharwal**
**What kind of change does this PR introduce?**
      bugfix, feature

**Issue Number:**

Fixes #2270

**Did you add tests for your changes?**

No

**Snapshots/Videos:**

Screenshot 2024-12-08 183611

**If relevant, did you update the documentation?**

**Summary**
This PR addresses the following issues:

- Set up a script to start Docker and import sample data.
- Enabled accessible ports for the localhost.
- Ensured containers can now communicate with each other.

These changes improve the developer experience by simplifying the environment setup and ensuring seamless container interactions.

**Does this PR introduce a breaking change?**
No 

**Other information**

None

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
	- Introduced a new collection for venues, including details for four distinct venues.
	- Enhanced data handling capabilities to support the new venue collection.
	- Improved setup process for Docker, including user prompts and MongoDB readiness checks.

- **Bug Fixes**
	- Updated volume mapping for the Caddy service to improve configuration loading.

- **Documentation**
	- Updated interfaces and enums to include the new "Venue" value.

- **Chores**
	- Modified utility scripts to incorporate the new venue model for data insertion and removal processes.
	- Updated test cases for volunteer ranks to reflect correct expected values.

### CODE REVIEW

File: docker-compose.dev.yaml [hunk 1]
```diff
@@ -57,7 +57,7 @@ services:
       - "80:80"
       - "443:443"
     volumes:
-      - $PWD/Caddyfile:/etc/caddy/Caddyfile
+      - ./Caddyfile:/etc/caddy/Caddyfile
```
[the remark this answers is not in the document]
Comment (VanshikaSabharwal) [reply docker-compose.dev.yaml:60]: the while does not work with PWD. As there is not PWD variable in .env

File: src/utilities/loadSampleData.ts [hunk 2]
```diff
@@ -201,6 +206,7 @@ const collections = [
   "organizations",
   "posts",
   "events",
+  "venue",
```
[the remark this answers is not in the document]
Comment (VanshikaSabharwal) [reply src/utilities/loadSampleData.ts:209]: did it

**Comment by palisadoes** [comment]
Reopening as an exception.

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by VanshikaSabharwal** [comment]
@palisadoes I fixed the failing test and this [test](https://github.com/PalisadoesFoundation/talawa-api/actions/runs/12257965904/job/34196657912?pr=2751) is failing due to changes i made in `docker-compose.dev.yaml` file. The change was necessary to make the ports accessible on local computer.

**Comment by palisadoes** [comment]
Does this work with any combination of docker and non docker implementations for Talawa API and Talawa Admin?

**Comment by VanshikaSabharwal** [comment]
> Does this work with any combination of docker and non docker implementations for Talawa API and Talawa Admin?

@palisadoes The system is being designed to support any combination of Dockerized and non-Dockerized setups for Talawa API and Talawa Admin. If both are Dockerized, a shared network is required for communication. While contributors can set this up manually, I’m working on automating the process for a smoother experience. The design also handles mixed environments (Dockerized and non-Dockerized) by ensuring proper connections and configurations, making the setup seamless for all contributors.

**Comment by palisadoes** [comment]
Is this ready to be merged?

**Comment by VanshikaSabharwal** [comment]
Yes @palisadoes

**Review by palisadoes — APPROVED** [review]

**Pull request merged**